src/Microsoft.DotNet.Darc/DarcLib/Architecture.md

```diff
@@ -0,0 +1,80 @@
+This document describes some of the architecture of the code in `DarcLib`, especially classes for managing git.
+
+### Git clients
+
+The `DarcLib` has several git-related classes which manage either a local git directory or a remote GitHub/AzureDevOps repository.
```

Comment:
Can you add one sentence explaining why we need several clients? This is the first question that any reader would have here. Cheers!

Reply:
Just pushed a commit

Reply:
Ah, will add one more

Reply:
@tkapin can you re-approve please?